octobot_trading/exchanges/exchange_simulator.py

```diff
@@ -61,14 +64,22 @@ def __init__(self, config, exchange_type, exchange_manager, backtesting_data_fil
         timestamps = [await importer.get_data_timestamp_interval()
                       for importer in self.exchange_importers]  # [(min, max) ... ]
 
-        await get_chan(OctoBotBacktestingChannelsName.TIME_CHANNEL.value).modify(
+        await modify_backtesting_timestamps(
+            self.backtesting,
             minimum_timestamp=min(timestamps)[0],
             maximum_timestamp=max(timestamps)[1])
 
     async def create_backtesting_exchange_producers(self):
         for importer in self.exchange_importers:
+            available_data_types = get_available_data_types(importer)
             for updater in UNAUTHENTICATED_UPDATER_SIMULATOR_PRODUCERS:
-                await updater(get_trading_chan(updater.CHANNEL_NAME, self.exchange_manager.id), importer).run()
+                if updater not in SIMULATOR_PRODUCERS_TO_DATA_TYPE \
+                        or any(required_data_type in available_data_types
+                               for required_data_type in SIMULATOR_PRODUCERS_TO_DATA_TYPE[updater]):
+                    if updater == MarkPriceUpdater:
```

Comment:
Why not creating a MarkPriceUpdaterSimulator ?

Reply:
Because it doesn't exist and would be a clone of MarkPriceUpdater, MarkPriceUpdater is a recent trades consumer

Reply:
I think it's better to have `MarkPriceUpdaterSimulator` that only herit from `MarkPriceUpdater` but with an `importer` attribute than having a ` if updater == MarkPriceUpdater` here.